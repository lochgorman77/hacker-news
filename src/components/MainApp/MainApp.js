import React, { Component } from 'react';
import { withStyles } from '@material-ui/core/styles';
import { Route } from 'react-router-dom';
import PropTypes from 'prop-types';

import TopStoriesContainer from 'components/TopStories/TopStoriesContainer';
import NewStoriesContainer from 'components/NewStories/NewStoriesContainer';
import HeaderMenu from 'components/Menu/HeaderMenu';

const styles = (theme) => ({
  root: {
    margin: 'auto',
    width: '90%',
  },
});

export class MainApp extends Component {
  render() {
    return (
      <section id="page-container" className={this.props.classes.root}>
        <HeaderMenu />

        <Route path="/topstories" component={TopStoriesContainer} />
        <Route path="/newstories" component={NewStoriesContainer} />
        <Route path="/" component={TopStoriesContainer} />
      </section>
    );
  }
}
MainApp.propTypes = {
  match: PropTypes.object,
  classes: PropTypes.object,
};
export const MainAppStyle = withStyles(styles, { name: 'MainApp' })(MainApp);
export default MainAppStyle;
