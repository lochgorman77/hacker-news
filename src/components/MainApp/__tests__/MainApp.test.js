import React from 'react';
import { MainAppStyle } from '../MainApp';

const props = {};

describe('<MainApp />', () => {
  let shallow, wrapper;

  beforeEach(() => {
    shallow = createShallow({ dive: true });
    wrapper = shallow(<MainAppStyle {...props} />);
  });

  it('renders correctly', () => {
    expect(wrapper.length).toBe(1);
    expect(wrapper.name()).toBe('section');
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
