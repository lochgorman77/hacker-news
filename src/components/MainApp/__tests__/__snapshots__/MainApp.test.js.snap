// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainApp /> renders correctly 1`] = `
<section
  className="MainApp-root"
  id="page-container"
>
  <HeaderMenu />
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "type": [Function],
      }
    }
    path="/topstories"
  />
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "type": [Function],
      }
    }
    path="/newstories"
  />
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "type": [Function],
      }
    }
    path="/"
  />
</section>
`;
